ugs-core/src/com/willwinder/universalgcodesender/gcode/processors/MeshLeveler.java

```diff
@@ -155,17 +153,23 @@ public List<String> processCommand(final String commandString, GcodeState state)
 
         GcodeMeta command = commands.get(0);
 
-        if (command == null || command.point == null) {
-            throw new GcodeParserException(ERROR_MISSING_POINT_DATA);
+        if (command == null) {
+            throw new GcodeParserException(ERROR_MISSING_POINT_DATA + commandString);
+        }
+        if (command.point == null) {
+            return Collections.singletonList(commandString);
+        }
+
+        // If we're in relative mode there is no adjustment needed
+        // TODO: This won't be true if we start in relative mode, in which case we need to ensure the first command
+        // is to move down by the offset at (0, 0)
+        if (!state.inAbsoluteMode) {
+            return Collections.singletonList(commandString);
```

Comment:
Is this really true? If there is a difference in the mesh heights between the current point and the previous one doesn't it need to be calculated and applied?

Reply:
in other words, the current `zPointOffset` minus the previous `zPointOffset`

Reply:
You're right. The previous code was wrong, but so is this. I'll send out an updated patch.

Reply:
Updated. I was previously testing with gcode that only had relative z axis-only moves. This time I tested with the gcode below. It does still have an issue when the first move is relative, but I think it requires a broader change to be able properly detect and deal with this situation.

```
G91
G1 Z-1F300
G1 X10Z2F1000
G1 Y10Z2
G1 X-10Z-2
G1 Y-10Z-2
```